src/legacy/ui/public/indexed_array/index.d.ts

```diff
@@ -32,6 +32,9 @@ declare class IndexedArray<T> extends Array<T> {
   public raw: T[];
   // May not actually be present, is dynamically defined.
   public inOrder: T[];
+  public byValue: {
```

Comment:
This is a dynamic function name depending on `index` value. It can be `byVal`, `byValue`, `byName` and so on.

Reply:
Could you propose a way to handle dynamic keys ? I can't find one.. 😢

Reply:
As an option you can extend this type for `options` in `SelectParamEditprProps` and add `byValue` to new type:
```
type SelectOptions = IndexedArray<SelectValueProp> & {
  byValue: { [key: string]: SelectValueProp };
}
```

I think that it's relevant because `options` was created with `index: ['value']` https://github.com/elastic/kibana/pull/35134/files?file-filters%5B%5D=.js&file-filters%5B%5D=.ts#diff-927a2f5054bd43b46c4965007d70b774R29